lldb/source/Plugins/Platform/MacOSX/objcxx/PlatformiOSSimulatorCoreSimulatorSupport.mm

```diff
@@ -399,18 +399,18 @@ static Status HandleFileAction(ProcessLaunchInfo &launch_info,
     case FileAction::eFileActionOpen: {
       FileSpec file_spec = file_action->GetFileSpec();
       if (file_spec) {
-        const int primary_fd = launch_info.GetPTY().GetPrimaryFileDescriptor();
+        const int primary_fd = launch_info.GetPTY()->GetPrimaryFileDescriptor();
```

Comment:
This looks like a worse API than before. The call sites now look like they forgot to check for a nullptr.
Could we keep the old API and have some callers use `shared_froom_this()`? Or have both APIs?